scripts-vagrant_provision/provision.yml

```diff
@@ -0,0 +1,175 @@
+      file:
+        dest: "/opt"
+        owner: vagrant
+        group: vagrant
+        mode: g=u
+        recurse: yes
+
+    - name: Create directories and set permissions
+      file:
+        path: "{{ item }}"
+        state: directory
+        owner: vagrant
+        group: vagrant
+        recurse: yes
+      loop:
+        - /opt/myaegee
+        - /opt/backups
+
+    - name: Check current Nodejs version
+      shell: node --version | tr -d 'v'
+      register: current_node_version
+      ignore_errors: yes
+
+    - name: Make sure there's no Node.js and npm
+      apt:
+        name: "{{ item }}"
+        state: absent
+      loop:
+        - node
+        - nodejs
+        - npm
+      when: current_node_version.stdout != nodejs_version
+
+- name: Install pyenv with better role but possibly with a deprecation inside
+  #TODO: pyenv role uses deprecated stuff?
+  hosts: all
+  tags: ['pyenv']
+
+  vars:
+    python_version: "3.10.3"
+
+  pre_tasks:
+    - name: Check current python version
+      shell: python --version | sed 's/Python //'
+      register: current_python_version
+      ignore_errors: yes
+
+  roles:
+    - role: avanov.pyenv
+      vars:
+        pyenv_path: "{{ ansible_env.HOME }}/.pyenv"
+        pyenv_python_versions: [ "{{ python_version }}" ]
+        pyenv_global: "{{ python_version }}"
+        pyenv_virtualenvs:
+          - venv_name: "notifications"
+            py_version: "{{ python_version }}"
+      when: current_python_version.stdout != python_version
+
+  # post_tasks:
+
+  #   - name: Install Python packages from requirements file
+  #     pip:
+  #       requirements: /vagrant/scripts-vagrant_provision/requirements.txt
+
+- name: Prerequisites - node/python itself
+  hosts: all
+  become: no #role asked so
+  tags: ['node']
+
+  vars:
+    nvm_commands:
+      - npm install -g npm@{{ npm_version }}
+
+  roles:
+    - role: morgangraphics.ansible_role_nvm
+      become: true
+      become_user: vagrant
+
+###
+- name: Install docker
+  hosts: all
+  become: true
+  tags: ['docker']
+
+  vars:
+    docker_packages:
+      - "docker-{{ docker_edition }}={{docker_ubuntu_version}}"
+      - "docker-{{ docker_edition }}-cli={{docker_ubuntu_version}}"
+      #- "docker-{{ docker_edition }}-rootless-extras={{docker_ubuntu_version}}"
+    docker_daemon_options:
+      log-driver: "json-file"
+      log-opts:
+        max-size: "10m"
+        max-file: "2"
+        labels: "host-{{ ansible_hostname }}"
+
+  roles:
+    - geerlingguy.docker
+
+###
+- name: Install last bunch of packages
+  hosts: all
+  become: true
+  tags: ['packages']
+
+  tasks:
+
+    - name: Install other required packages
+      apt:
+        name: "{{ item }}"
+        state: present
+      loop: "{{ packages_to_install }}"
+
+###
+- name: Create banners for connection
+  hosts: all
+  become: true
+  tags: ['motd']
+
+  tasks:
+      - name: Ensure a locale exists
+        locale_gen:
+          name: en_IE.UTF-8
```

Comment:
Not that it really matters, but why Ireland?

Reply:
it's the locale I have, US is shit and GB has something that I don't remember (the reason I have is a bonus for italian speakers: gaelic has accents éáó so I can use them)